File: Week1/1-2_최윤성_20210705.cpp
#include <string>
#include <vector>

using namespace std;

int solution(string skill, vector<string> skill_trees) {
    int answer = 0;
    
    for(string tree : skill_trees) {
        bool able = true;
        for(int i = skill.length(); i >= 0; i--) {
            int index = tree.find(skill[i]);
            if(index != string::npos && i > 0)
                if(tree.substr(0, index).find(skill[i - 1]) == string::npos)
                    able = false;
        }
        if(able) answer++;
    }
    
    return answer;
}

File: Week1/1-2_최윤성_20210705.py
def solution(skill, skill_trees):
    answer = 0
    
    for tree in skill_trees:
        able = True
        for i in range(len(skill) - 1, 0, -1):
            index = tree.find(skill[i])
            if index != -1 and i > 0:
                if tree[:index].find(skill[i - 1]) == -1:
                    able = False
                    break
        if able: 
            answer += 1
                
    return answer

if __name__ == "__main__":
    skill = "CBD"
    skill_trees	= ["BACDE", "CBADF", "AECB", "BDA"]	
    solution(skill=skill, skill_trees=skill_trees)